### ISSUE TITLE
Add new field for Swarm in Docker Engine API

### ISSUE THREAD

**Issue by nishanttotla** [docker/engine-api#44]
Currently, the `/info` endpoint for the Swarm API uses `DriverStatus` field in the Engine API. We would like to create a new field exclusively for Swarm to use.

The current suggestion is to name the field `CustomStatus`.

Related issues: https://github.com/docker/docker/issues/19277, https://github.com/docker/swarm/issues/1625, https://github.com/samalba/dockerclient/issues/215

cc @calavera @vieux

**Comment by vieux** [comment]
:+1:

**Comment by dnephin** [comment]
`CustomStatus` is not very descriptive. Wouldn't it make sense to use something more descriptive about the values it contains? I assume it's about the node status, so maybe `NodeStatus`? or `ClusterStatus` ?

**Comment by nishanttotla** [comment]
@dnephin The idea is that Docker Engine isn't tied to Swarm only, so we want a more generic "extra info" field that could be used by other tools also. What do you think?

**Comment by dnephin** [comment]
I think it would still be better to be descriptive, engine can just return nil for the value and it will be omitted from the response.

**Comment by nishanttotla** [comment]
Fair point, but it bloats the API, and depending on how Swarm evolves, may not be needed later. A general `[][2]string` field on the other hand will be more flexible.

@calavera what do you think?

**Comment by dongluochen** [comment]
@nishanttotla Can you provide examples how this field will be used?

**Comment by nishanttotla** [comment]
The current use case is only for Swarm (as per this issue https://github.com/docker/swarm/issues/1625). An example for what this field might contain is as follows:

```
[][]string{
{"Role", "primary"},
{"Strategy", "spread"},
{"Nodes", "2"},
{" agent-01", "x.x.x.x:2375"},
{" └ Status", "Healthy"},
...
}
```
Essentially the entire cluster-related information.

**Comment by calavera** [comment]
This has been fixed in #46
